lib/synapse/haproxy.rb

```diff
@@ -741,6 +742,10 @@ def generate_backend_stanza(watcher, config)
           backend = backends[backend_name]
           b = "\tserver #{backend_name} #{backend['host']}:#{backend['port']}"
           b = "#{b} cookie #{backend_name}" unless config.include?('mode tcp')
+          if !@opts['ignore_weights'] && backend.has_key?('weight')
+            weight = backend['weight'].to_i
+            b = "#{b} weight #{weight}"
```

Comment:
So I believe that HAProxy will use the last weight provided. If that's true I think this should probably slot this in between the HAProxy backend `server_options` and the server `haproxy_server_options` level.